src/connectors/autocomplete/connectAutocomplete.ts

```diff
@@ -0,0 +1,188 @@
+
+const connectAutocomplete: AutocompleteConnector = (
+  renderFn,
+  unmountFn = noop
+) => {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const { escapeHTML = true } = widgetParams || {};
+
+    warning(
+      !(widgetParams as any).indices,
+      `
+The option \`indices\` has been removed from the Autocomplete connector.
+
+The indices to target are now inferred from the widgets tree.
+${
+  Array.isArray((widgetParams as any).indices)
+    ? `
+An alternative would be:
+
+const autocomplete = connectAutocomplete(renderer);
+
+search.addWidgets([
+  ${(widgetParams as any).indices
+    .map(({ value }: { value: string }) => `index({ indexName: '${value}'}),`)
+    .join('\n  ')}
+  autocomplete()
+]);
+`
+    : ''
+}
+      `
+    );
+
+    type ConnectorState = {
+      instantSearchInstance?: InstantSearch;
+      refine?: (query: string) => void;
+    };
+
+    const connectorState: ConnectorState = {};
+
+    return {
+      $$type: 'ais.autocomplete',
+
+      getConfiguration(previousParameters) {
+        const parameters = {
+          query: previousParameters.query || '',
+        };
+
+        if (!escapeHTML) {
+          return previousParameters.setQueryParameters(parameters);
+        }
+
+        return previousParameters.setQueryParameters({
+          ...parameters,
+          ...TAG_PLACEHOLDER,
+        });
+      },
+
+      init({ instantSearchInstance, helper }) {
+        connectorState.instantSearchInstance = instantSearchInstance;
+        connectorState.refine = (query: string) => {
+          helper.setQuery(query).search();
+        };
+
+        renderFn(
+          {
+            widgetParams,
+            currentRefinement: helper.state.query || '',
+            indices: [],
+            refine: connectorState.refine,
+            instantSearchInstance: connectorState.instantSearchInstance,
+          },
+          true
+        );
+      },
+
+      render({ helper, scopedResults }) {
+        const indices = scopedResults.map(scopedResult => {
+          // We need to escape the hits because highlighting
+          // exposes HTML tags to the end-user.
+          scopedResult.results.hits = escapeHTML
+            ? escapeHits(scopedResult.results.hits)
+            : scopedResult.results.hits;
+
+          return {
+            indexName: scopedResult.results.index,
+            hits: scopedResult.results.hits,
+            results: scopedResult.results,
+          };
+        });
+
+        renderFn(
+          {
+            widgetParams,
+            currentRefinement: helper.state.query || '',
+            indices,
+            refine: connectorState.refine!,
+            instantSearchInstance: connectorState.instantSearchInstance!,
+          },
+          false
+        );
+      },
+
+      dispose({ state }) {
+        unmountFn();
+
+        const stateWithoutQuery = state.setQueryParameter('query', undefined);
```

Comment:
This is fine, but we should probably avoid calling state functions multiple times, and can join them all in a single setQueryParameters, WDYT?

Reply:
We do that in other connectors. They don't mutate the global state, they return a new object.
Why would you recommend not calling these functions multiple times?

Reply:
Here it's to avoid the duplication of the logic for removing the query. At some point if we have perf issue we could revisit those parts, but for now I guess it's fine.